src/Grpc.Net.Client/GrpcChannelOptions.cs

```diff
@@ -65,6 +66,56 @@ public sealed class GrpcChannelOptions
         /// </summary>
         public int? MaxReceiveMessageSize { get; set; }
 
+        /// <summary>
+        /// Gets or sets the maximum retry attempts. This value limits any retry and hedging attempt values specified in
+        /// the service config.
+        /// <para>
+        /// Setting this value alone doesn't enable retries. Retries are enabled in the service config, which can be done
+        /// using <see cref="ServiceConfig"/>.
+        /// </para>
+        /// <para>
+        /// A <c>null</c> value removes the maximum retry attempts limit. Defaults to 5.
+        /// </para>
+        /// <para>
+        /// Note: Experimental API that can change or be removed without any prior notice.
+        /// </para>
+        /// </summary>
+        public int? MaxRetryAttempts { get; set; }
+
+        /// <summary>
+        /// Gets or sets the maximum buffer size in bytes that can be used to store sent messages when retrying
+        /// or hedging calls. If the buffer limit is exceeded then no more retry attempts are made and all
+        /// hedging calls but one will be canceled. This limit is applied across all calls made using the channel.
+        /// <para>
```

Comment:
> all hedging calls but one will be canceled.

Why not allow as many hedging calls as can fit under the limit?

Reply:
If you're making a unary call (one message that exceeds limit) then other hedging calls haven't started.

If you're making a client streaming call (a new message has exceeded the limit) then the other hedging calls won't have all the client messages. They need to be canceled.